x-pack/solutions/observability/plugins/apm/server/routes/fleet/api_keys/add_api_keys_to_policies_if_missing.ts

```diff
@@ -19,26 +22,45 @@ export async function addApiKeysToEveryPackagePolicyIfMissing({
   coreStartPromise,
   pluginStartPromise,
   logger,
+  licensing,
 }: {
   coreStartPromise: Promise<CoreStart>;
   pluginStartPromise: Promise<APMPluginStartDependencies>;
   logger: Logger;
+  licensing: LicensingPluginSetup;
 }) {
   const coreStart = await coreStartPromise;
   const { fleet } = await pluginStartPromise;
   if (!fleet) {
     return;
   }
 
+  // We need to wait for the licence feature to be available,
+  // to have our internal saved object client with encrypted saved object working properly
+  await licensing.license$
+    .pipe(
+      filter(
+        (licence) =>
+          licence.getFeature('security').isEnabled && licence.getFeature('security').isAvailable
+      ),
+      take(1)
+    )
+    .toPromise();
```

Comment:
Isn't `toPromise` this deprecated? you could probably replace it with `lastValueFrom`